src/arith/rewrite_simplify.cc

```diff
@@ -123,7 +123,7 @@ PrimExpr RewriteSimplifier::Impl::VisitExpr_(const AddNode* op) {
   // Pattern var to match any expression
   PVar<PrimExpr> x, y, z, b1, b2, s1, s2;
   // Pattern var match IntImm
-  PVar<IntImm> c1, c2, c3;
+  PVarOpt<Optional<IntImm>> c1, c2, c3;
```

Comment:
I don't think it is right to use optional here, as the intention is to match the variable itself

Reply:
Yes, I agree. It is only to accommodate absent of default constructor. However the behaviour don't violate with PVar I think.

Reply:
I agree too...Is there any better to do this?

Reply:
I have 2 possible approach here:

A0: Rename PVarOpt --> PVarExt(Extended) and hide Optional<T> input format.  
Look like --> `PVarExt<IntImm> c1, c2, c3;`

A1: Provide default constructor with default values. 
Look like: `PVar<IntImm> c1(IntImm(DataType::Int(32), 0)), c2(IntImm(DataType::Int(32), 0)), c3(IntImm(DataType::Int(32), 0));`

Please let me know your thoughts on this. Thanks!

Reply:
I don;t think either approaches are good here. Let us leave it as it is for now